### ISSUE TITLE
Version conflicts and failing tests

### ISSUE THREAD

**Issue by ernestum** [HumanCompatibleAI/imitation#398]
When I install the cloned repository with "test" extras, I get the following error:

`ERROR: stable-baselines3 1.4.0 has requirement gym<0.20,>=0.17, but you'll have gym 0.21.0 which is incompatible.`

Also I can't run the tests since ray seems to be missing.

Steps to reproduce:
```bash
git clone git@github.com:HumanCompatibleAI/imitation.git
cd imitation
python3 -m venv venv
source venv/bin/activate
pip install ".[test]"
pytest
```
After installing with `docs` and `parallel` extras as done in the `ci/build_venv.sh` script (`pip install ".[docs,parallel,test]"`) ray is found but the tests still fail.

Again deducing from the `ci/build_venv.sh` that the mujoco envs might be required I installed it according to the instructions. (`apt install patchelf build-essential libgl1-mesa-dev libgl1-mesa-glx libglew-dev` helped).
Tests still fail.

By far the most common failed assertion was during loading of a PPO model in sb3:
```python
venv/lib/python3.8/site-packages/imitation/policies/serialize.py:122: in load_policy
    return agent_loader(policy_path, venv)
venv/lib/python3.8/site-packages/imitation/policies/serialize.py:77: in f
    model = load_stable_baselines_model(cls, path, venv)
venv/lib/python3.8/site-packages/imitation/policies/serialize.py:60: in load_stable_baselines_model
    return cls.load(model_path, env=venv, **kwargs)
venv/lib/python3.8/site-packages/stable_baselines3/common/base_class.py:730: in load
    model._setup_model()
venv/lib/python3.8/site-packages/stable_baselines3/ppo/ppo.py:158: in _setup_model
    self.clip_range = get_schedule_fn(self.clip_range)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

value_schedule = deque([])

    def get_schedule_fn(value_schedule: Union[Schedule, float, int]) -> Schedule:
        """
        Transform (if needed) learning rate and clip range (for PPO)
        to callable.
    
        :param value_schedule:
        :return:
        """
        # If the passed schedule is a float
        # create a constant function
        if isinstance(value_schedule, (float, int)):
            # Cast to float to avoid errors
            value_schedule = constant_fn(float(value_schedule))
        else:
>           assert callable(value_schedule)
E           AssertionError
```
My suspicion right now is, that the saved expert models in `tests/testdata/expert_models` were saved using an older sb3 version in a format that is not compatible with the current version?

I attached the `pip freeze` for reference:
[requirements.txt](https://github.com/HumanCompatibleAI/imitation/files/7985975/requirements.txt)

**Comment by AdamGleave** [comment]
We should add `ray` to the `test` requirement, thanks for catching that.

I run into `Box` not having `shape` objects for `gym==0.19`, so I think we do actually need gym `0.20` or later. We could downgrade and modify but I'd rather not as we will want to upgrade to gym `0.22` once it's out and SB3 merges https://github.com/DLR-RM/stable-baselines3/pull/734 I've not seen any issues with SB3 and latest gym version, I think it's just a problem with Atari (which our algorithms mostly don't work with anyway...)

Which test case is failing? All tests are passing for me
[versions.txt](https://github.com/HumanCompatibleAI/imitation/files/7988217/versions.txt)

**Comment by AdamGleave** [comment]
Fixed in #400
